snuba/query/processors/granularity_processor.py

```diff
@@ -31,7 +34,20 @@ def __get_granularity(query: Query) -> int:
         )
 
     def process_query(self, query: Query, request_settings: RequestSettings) -> None:
-        granularity = self.__get_granularity(query)
+        granularity = None
+        if isinstance(request_settings, SubscriptionRequestSettings):
+            from_date, to_date = get_time_range(query, "timestamp")
```

Comment:
Even though this processor is currently only used in metrics, you could fetch the time column programmatically so that you don't have to hardcode it. See my example below (you might have to add extra type checking).


```suggestion
            entity = get_entity(query.get_from_clause().key)
            from_date, to_date = get_time_range(query, entity.required_time_column)
```

Reply:
sounds good